### PR TITLE
[tests-only][full-ci] Improve Then step in apiShareManagementToRoot suites

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
This PR adds `Then` steps to existing API test scenarios to better check the results of the `When` steps.

**Suites covered:**
- `apiShareManagementToRoot`

## Related Issue

- Part of https://github.com/owncloud/core/issues/39512

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/apiShareManagementToRoot/mergeShare.feature [hunk 1]
```diff
@@ -84,13 +100,18 @@ Feature: sharing
...
+    And as user "Brian" folder "/merge-test-outside-groups-renamebeforesecondshare-renamed" should contain a property "oc:permissions" with value "SRDNVCK"
     And as "Brian" folder "/merge-test-outside-groups-renamebeforesecondshare" should not exist
 
   Scenario: Merging shares for recipient when shared from outside with user then group and recipient renames in between
     Given user "Alice" has created folder "/merge-test-outside-groups-renamebeforesecondshare"
     When user "Alice" shares folder "/merge-test-outside-groups-renamebeforesecondshare" with user "Brian" using the sharing API
     And user "Brian" moves folder "/merge-test-outside-groups-renamebeforesecondshare" to "/merge-test-outside-groups-renamebeforesecondshare-renamed" using the WebDAV API
     And user "Alice" shares folder "/merge-test-outside-groups-renamebeforesecondshare" with group "grp1" using the sharing API
-    Then as user "Brian" folder "/merge-test-outside-groups-renamebeforesecondshare-renamed" should contain a property "oc:permissions" with value "SRDNVCK"
+    Then the HTTP status code of responses on each endpoint should be "200,201,200" respectively
```
Comment (SagarGi) [inline tests/acceptance/features/apiShareManagementToRoot/mergeShare.feature:114]:
```suggestion
    Then the HTTP status code of responses on each endpoint should be "200, 201, 200" respectively
```

File: tests/acceptance/features/apiShareManagementToRoot/mergeShare.feature [hunk 2]
```diff
@@ -84,13 +100,18 @@ Feature: sharing
     When user "Alice" shares folder "/merge-test-outside-groups-renamebeforesecondshare" with group "grp1" using the sharing API
     And user "Brian" moves folder "/merge-test-outside-groups-renamebeforesecondshare" to "/merge-test-outside-groups-renamebeforesecondshare-renamed" using the WebDAV API
     And user "Alice" shares folder "/merge-test-outside-groups-renamebeforesecondshare" with user "Brian" using the sharing API
-    Then as user "Brian" folder "/merge-test-outside-groups-renamebeforesecondshare-renamed" should contain a property "oc:permissions" with value "SRDNVCK"
+    Then the HTTP status code of responses on each endpoint should be "200,201,200" respectively
```
Comment (SagarGi) [inline tests/acceptance/features/apiShareManagementToRoot/mergeShare.feature:103]:
```suggestion
    Then the HTTP status code of responses on each endpoint should be "200, 201, 200" respectively
```

**Review by SagarGi — COMMENTED** [review]
Others LGTM :+1:

**Comment by kiranparajuli589** [comment]
lgtm :+1:

**Comment by phil-davis** [comment]
@amrita-shrestha please rebase - GitHub reports conflicts

**Review by phil-davis — APPROVED** [review]

**Pull request merged**